packer/plugin.go

```diff
@@ -248,6 +269,13 @@ func (c *PluginConfig) discoverSingle(glob string) (map[string]string, error) {
 
 		// Look for foo-bar-baz. The plugin name is "baz"
 		pluginName := file[len(prefix):]
+
+		// if Plugin name has OS_ARCH in it, split at _v(0-9) with regex
+		if strings.Contains(pluginName, OS_ARCH) {
```

Comment:
To be a bit more specific with the match you can use `strings.HasSuffix`. 
```suggestion
		if strings.HasSuffix(pluginName, OS_ARCH) {
```

Since OS_ARCH is part of the glob I wonder if we could just check if the name contains an understood to determine if we need to split. I don't think that plugins names such as `packer-plugin-mycould_rocks_v1.0.0_x5.0_OS_ARCH` or `packer-plugin-yourcloud_rocks` will ever be value.